jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarExtractor.java

```diff
@@ -72,6 +74,18 @@ public static void extract(Path source, Path destination) throws IOException {
           }
         }
       }
+
+      // Preserve source modification timestamps of files. If the entry is a symbolic link then set
+      // it's modification time to that of the target file.
+      for (TarArchiveEntry entry : modificationTimeMap.keySet()) {
+        FileTime sourceModificationTime = modificationTimeMap.get(entry);
+        if (entry.isSymbolicLink()) {
+          Path targetPath = destination.resolve(entry.getName()).toRealPath();
+          Path targetRelativePath = destination.relativize(targetPath);
+          sourceModificationTime = modificationTimeMap.get(targetRelativePath);
```

Comment:
I don't have confidence if this computation guarantees reconstructing the original path. (Path computation is difficult and depends on many factors. Honestly I don't know what `toRealPath()` does honestly, its Javadoc also says "the precise definition of this method is implementation dependent.") I also don't really like creating a map structure for this. Can't we just set timestamps inside the loop above?

Reply:
> I don't have confidence if this computation guarantees reconstructing the original path. (Path computation is difficult and depends on many factors. Honestly I don't know what `toRealPath()` does honestly, its Javadoc also says "the precise definition of this method is implementation dependent.")

Ah thank you for catching this! I looked into this a bit more and I think `Files.readSymbolicLink` is what I need to get the target of the symbolic link.

> I also don't really like creating a map structure for this. Can't we just set timestamps inside the loop above?

Calling set timestamp inside the for loop for the symlink was initially resulting in a `FileNotFoundException`. This was because we were calling `setLastModifiedTime` *before* the target file was created. The call to `setLastModifiedTime` on a symlink worked fine if the target file was created before the symlink but not the other way around. This is why I had to place the code at the end of the method (after all the files has been created).

Reply:
> to get the target of the symbolic link.

We shouldn't be concerned about the target? Are you trying to modify the target? Also, the target may not actually exist in an archive (i.e., a broken symlink). This shouldn't cause a problem.